### ISSUE TITLE
Redefining variable type

### ISSUE THREAD

**Issue by jayvdb** [adsharma/py2many#287]
The following fails for at least Dart, but likely most other languages.
```py
a = 1
a = 1.0
```

**Comment by adsharma** [comment]
Yes - this is legal python, but not legal in a statically typed language.
I'm thinking we should have a linter which gives actionable feedback, so it could be rewritten as:

```
a = 1
a_float = 1.0
```

This is the crux of https://github.com/adsharma/py2many/issues/260

The py2many approach is to sacrifice compatibility for lower complexity and overall performance. I'm not sure if the broader python community accepts it.

**Comment by jayvdb** [comment]
Until an implementation is decided upon, it should trigger a `NotImplementedError`
